minidns-core/src/main/java/de/measite/minidns/util/Base32.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.util;
+
+/**
+ * Very minimal Base32 encoder.
+ */
+public final class Base32 {
+    private static final String ALPHABET = "0123456789ABCDEFGHIJKLMNOPQRSTUV";
+    private static final String PADDING = "======";
+
+    /**
+     * Do not allow to instantiate Base32
+     */
+    private Base32() {
+    }
+
+    public static String encodeToString(byte[] bytes) {
+        int paddingCount = (int) (8 - (bytes.length % 5) * 1.6) % 8;
```

Comment:
I find this really hard to read. It could be written as
```
final static int[] PADDING_LUT = new int[]{0,6,4,3,1};
[...]
paddingCount = PADDING_LUT[bytes.length % 5];
```